### ISSUE TITLE
Dank UI - Discussion

### ISSUE THREAD

**Issue by dotvhs** [Tunous/Dawn#7]
First of all, I'd like to say that I do really like Dank in general, in terms of UI and the feeling. However, there are few things I think might or even should be changed for more consistency and better UX. So I created this thread for discussion.

Something I think I know how should be fixed:
![image](https://user-images.githubusercontent.com/681416/51003925-e2dd0c00-1538-11e9-9443-d962475a3c27.png)
Number of upvotes - while it is something that makes Dank stand out from the crowd, I do think that number should be shown after the title, not before it, at least as an option. Here's an example:
![image](https://user-images.githubusercontent.com/681416/51003938-eff9fb00-1538-11e9-8732-dba7baaecbc8.png)

It will be easier to read the titles that don't have similar score number (my eyes won't have to jump left/right anymore, to read the title). Here's an example that shows what makes me feel uncomfortable reading the titles:
![image](https://user-images.githubusercontent.com/681416/51004158-cdb4ad00-1539-11e9-8265-c5afd101b68a.png)

Another thing on the same screen is that horizontal bar with sorting button. I do like how this button looks like, but I don't like that it is alone, making almost whole line of wasted space and it makes no sense that it's fixed to that view:
![image](https://user-images.githubusercontent.com/681416/51003962-04d68e80-1539-11e9-8eeb-46abd4edebb5.png)
![image](https://user-images.githubusercontent.com/681416/51003980-14ee6e00-1539-11e9-89bb-2b956b5dcc8e.png)

What's the point of Toolbar having a shadow when it's not really covering anything since that grey bar with button that changes sorting is not moving at all. I honestly don't have a good idea what could be done for it to make more sense for now.

The last thing I find a bit annoying is way more minor than the rest. In comments view when there's picture on top of the screen, there is also close button in the left corner and of course there should be icon to close but sometimes it's obscuring something which does not really look good, here's an example:
![screenshot_20190111-001650](https://user-images.githubusercontent.com/681416/51003797-6e09d200-1538-11e9-875d-d55656a42d35.png)

I hope I did not sound harsh nor rude, I do have issues with explaining things in a gentle manner in foreign language. Sorry if I did.

**Comment by Esselans** [comment]
I didn't understand quite well the 'Toolbar shadow' issue but +1 to move the vote count and the close icon on the image.

Also, I notice there is no timestamp (i.e. '2 days ago' on the items list)

**Comment by voythas** [comment]
Ah yeah, I wrote that very poorly, so I will rephrase that:
Top part of app contains two things, a toolbar (that darker part with app/subreddit name etc) and that bar underneath it which contains only the button for switching sort type.

And both of them are fixed in place - they are pinned to that position, does not matter if you scroll - it's still on top of the screen. I asked what's the point of the shadow underneath the toolbar when toolbar itself is on top of the item that never moves and shadow on elements like this is used to give a sense of feeling that it is floating on top of other elements.

In short words - I don't like that there is quite big part of the screen pinned with quite a lot of space wasted.

**Comment by Esselans** [comment]
Ok maybe the sort button should be part of the scrollable content. Will that work according to your designs?

**Comment by saket** [comment]
I like the idea of moving the scores to the end. I kept them on the left so that my eyes could scan the scores easily. 

Any better ideas on improving the sorting button? I don't like its placement either, but couldn't come up with a better design.

**Comment by Esselans** [comment]
> Any better ideas on improving the sorting button? I don't like its placement either, but couldn't come up with a better design.

Well since the refresh button is in the upper toolbar we can move that button next to the sort button or move the sort button to the other toolbar. In other words, put the button that are related together.

Another option for me is to keep that bar (the one below the main toolbar) and put all the subreddit options there, like sort, (un)subscribe, refresh, submit link/post.

**Comment by saket** [comment]
> And both of them are fixed in place - they are pinned to that position, does not matter if you scroll - it's still on top of the screen. I asked what's the point of the shadow underneath the toolbar when toolbar itself is on top of the item that never moves and shadow on elements like this is used to give a sense of feeling that it is floating on top of other elements.

The initial plan was to hide either both of them or just the sorting bar when the content list is scrolled.

> Well since the refresh button is in the upper toolbar we can move that button next to the sort button or move the sort button to the other toolbar. In other words, put the button that are related together.

There's very little space on the toolbar. Subreddits with long name get truncated very easily. Additionally, there's the subscribe button also sits next to the sorting button.

**Comment by lastweakness** [comment]
@saket why not make the bar with sort a scrollable view with buttons like wiki, subscribe, etc?

**Comment by saket** [comment]
@lastweakness are you talking about the bar below the subreddit name?

**Comment by lastweakness** [comment]
@saket yes

**Comment by saket** [comment]
That's a good idea. Will do that once I add more options to the bar. Thanks!

**Comment by elenmerbau** [comment]
May I suggest remembering sorting order per subreddit as well, currently it just resets everything to "Best" every time I relaunch the app.

**Comment by elenmerbau** [comment]
I think it would be nice to have more colors, for example: having subreddits colored on the frontpage/popular to differentiate them more easily. 
Another suggestion is to hide "R/" from subreddit names to save some space.
![Screenshot_20191111-203128](https://user-images.githubusercontent.com/23418072/68595343-7669cb00-04c3-11ea-814a-85e551c290de.png)

**Comment by elenmerbau** [comment]
Found myself missing pressing back button 2 times in a row to close the app, sometimes I close it accidentally.

**Comment by Tunous** [comment]
Hey @elenmerbau thank you for the comments. Would you mind creating a new issue for each suggestion to make them more focused and easier to find in the future? That would help a lot.

**Comment by elenmerbau** [comment]
@Tunous aight, no problems, will do after I finish cookong :p

**Comment by Tunous** [comment]
Moved the missing topic to separate issue: https://github.com/Tunous/Dawn/issues/255 The other ones are in https://github.com/Tunous/Dawn/issues/208 and https://github.com/Tunous/Dawn/issues/119. Not sure about votes position but that perhaps will return to us somewhere :)

Closing this as I prefer to have more focused issues.
